block/manager.go

```diff
@@ -45,23 +47,23 @@ type newBlockEvent struct {
 // Manager is responsible for aggregating transactions into blocks.
 type Manager struct {
 	lastState types.State
+	// lastStateMtx is used by lastState
+	lastStateMtx *sync.Mutex
```

Comment:
i think it is better to use `RWMutex` with `RLock/RUnlock` for read cases and `Lock/Unlock` for writes. note that there are multiple writers (sync loop and aggr loop). `CreateBlock` only reads the last state and ApplyBlock writes to it. make sure to use RLock and Lock accordingly. also, saw that you have missed ApplyBlock in trySyncLoop.

Reply:
Thanks for the review. Addressed.